src/tests/unit/tests/DetailsView/components/save-assessment-button.test.tsx

```diff
@@ -50,100 +49,22 @@ describe('SaveAssessmentButton', () => {
             download: 'download',
             href: 'url',
             userConfigurationStoreData,
+            handleSaveAssessmentButtonClick: handleSaveAssessmentButtonClickMock.object,
         };
     });
 
-    describe('on dialog enabled', () => {
-        describe('render', () => {
-            beforeEach(() => {
-                mockReactComponent(Dialog, 'Dialog');
-                wrapper = render(<SaveAssessmentButton {...propsStub} />);
-                fireEvent.click(wrapper.container.querySelector('mock-customizedactionbutton'));
-            });
-            it('snapshot of dialog', () => {
-                expect(wrapper.asFragment()).toMatchSnapshot();
-            });
-
-            it('dialog is visible', () => {
-                expect(getMockComponentClassPropsForCall(Dialog, 2).hidden).toEqual(false);
-            });
-
-            it('dialog is hidden (dismissed) when onDismiss is called', () => {
-                act(() => {
-                    getMockComponentClassPropsForCall(Dialog, 2).onDismiss();
-                });
-                expect(getMockComponentClassPropsForCall(Dialog, 3).hidden).toEqual(true);
-            });
-        });
-        let wrapper: RenderResult;
-        describe('interaction', () => {
-            beforeEach(() => {
-                useOriginalReactElements('@fluentui/react', [
-                    'Dialog',
-                    'DialogFooter',
-                    'Stack',
-                    'Checkbox',
-                    'PrimaryButton',
-                    'ActionButton',
-                ]);
-
-                wrapper = render(<SaveAssessmentButton {...propsStub} />);
-                fireEvent.click(wrapper.getByRole('link'));
-            });
-
-            it('when "dont show again" box is clicked, set the showSaveAssessmentDialog user config state to `false`', () => {
-                // The "Don't show again" checkbox logic is inverted
-                const checkbox = wrapper.getByRole('checkbox');
-                // Check "Don't show again" = true
-
-                fireEvent.click(checkbox);
-                // showSaveAssessmentDialog = false ("Enable the dialog" = false)
-                userConfigMessageCreatorMock.verify(
-                    x => x.setSaveAssessmentDialogState(false),
-                    Times.atLeastOnce(),
-                );
-            });
-
-            it('should call saveAssessment on click', async () => {
-                assessmentActionMessageCreatorMock.verify(
-                    x => x.saveAssessment(It.isAny()),
-                    Times.atLeastOnce(),
-                );
-            });
-            it('dialog is hidden (dismissed) when "got it" button is clicked', async () => {
```

Comment:
this test is missing

Reply:
Yes I looked at the test code and the structure needs to change for this to work better. We need to put the interaction tests in a `describe` block that uses original components, sets `props.isOpen` to true, and mocks `props.onClose`. then we can check if `props.onClose` got called.

Reply:
Hi @madalynrose fixed this now, please re-review.